Update(Docs): Update intro page of CycloneDX (#141)
fossology/gsoc: docs/2023/cyclonedx/index.md

There are two main components of this project:

- First part is all about generating CycloneDx report in FOSSology. Now for report generation we need to follow these steps:

  - **Retrieve the data from the FOSSology DAO layer:** FOSSology has a database layer called the [DAO](https://github.com/fossology/fossology/tree/master/src/lib/php/Dao) (Data Access Object) layer that is used for accessing data from FOSSology's database. We can use the FOSSology DAO layer to fetch the required data from the database.
  - **Create a CycloneDX agent:** Develop a custom agent that interacts with the FOSSology DAO layer and retrieves the necessary data for generating the CycloneDX report. This agent will act as a bridge between FOSSology and the CycloneDX report generation process.
  - **Extract the required information:** Using the CycloneDX agent, retrieve the license and copyright information for each file from the FOSSology scan data. The specific information needed will depend on the requirements and the level of detail that needed to be include in the report.
  - **Implement a CycloneDX validator:** Develop a validator that can verify if a CycloneDX report adheres to the schema. 

>   - **Generate the CycloneDX report:** Once collected the relevant data from database and validate it to CycloneDx schema, we can generate the CycloneDX report.

Reviewer: ```suggestion
  - **Generate the CycloneDX report:** Once collected the relevant data from database with validated CycloneDX schema, we can generate the CycloneDX report.
```
